dashboard/utils.py

```diff
@@ -219,7 +229,7 @@ def has_paid_final_grade(self, edx_course_key):
         Returns:
             bool: whether a frozen final grade exists
         """
-        return self.final_grade_qset.paid_on_edx().for_course_run_key(edx_course_key).exists()
+        return self.has_final_grade(edx_course_key) and self.has_paid(edx_course_key)
```

Comment:
I'm guessing it's because a user can still be considered to have a paid final grade if (a) it's a FA program, (b) they have a FinalGrade that wasn't paid on edX, and (c) they _have_ paid via ecommerce. @giocalitri is that right?